mipengine/controller/node_tasks_handler_celery.py

```diff
@@ -118,45 +161,84 @@ def node_id(self):
     def node_data_address(self):
         return self._db_address
 
+    def _apply_async(self, task_signature, **kwargs) -> AsyncResult:
+        try:
+            # the reason the existing connection to the broker is passed here as an
+            # argument in apply_async is because the default behaviour (not passing a
+            # connection object), when the broker is down, is for the celery app to try to
+            # create a new connection to the broker, without raising any exceptions.
+            # Nevertheless while the broker is down the call to apply_async just hangs
+            # waiting for a connection with the broker to be established. Passing the
+            # existing connection object to apply_async causes the call to raise an
+            # exception if the broker is down
+            async_result = task_signature.apply_async(
```

Comment:
I removed task_to_async from celery_app.py and put it in the node_registry.py

Reply:
My idea was to have only one place where we have this "broker connection" fix. Right now it's implemented in 2 different places and I don't really see the reason why this can't be exported in a common method (DRY principle), especially because it's such a specific fix. Anyway, we disagree on the "celery_app" module so I won't push on this...